lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1520,12 +1528,18 @@ ChangeResult TypeAnalyzer::visitAtenTensorOp(AtenTensorOp op) {
 }
 
 template <typename OpTy>
-ChangeResult TypeAnalyzer::visitConstantTensorAllocOp(OpTy op) {
+ChangeResult
+TypeAnalyzer::visitConstantTensorAllocOp(OpTy op,
+                                         llvm::Optional<Type> inputDType) {
   auto knowledge =
       ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
   fillInSizesGivenSizesList(knowledge, op.size());
-  fillInDTypeGivenDTypeAndDataType(knowledge, op.dtype(),
-                                   Torch::FloatType::get(op->getContext()));
+  if (isa<AtenFullOp>(op))
+    fillInDTypeGivenDTypeIntAndInputDType(knowledge, op.dtype(),
+                                          inputDType.getValue());
+  else
```

Comment:
If you change the `inputDType` argument to `dataType` things can be simplified a little bit. One simplification would be that you wouldn't need this type conversion:

https://github.com/llvm/torch-mlir/blob/6256bf7405bc03aa6d99c5e4996efa67ef79c129/lib/Dialect/Torch/Transforms/RefineTypes.cpp#L386

The other would be this `if-else`, which can be simplified to:

```c++
if (!dataType)
  dataType = Torch::FloatType::get(op->getContext());
fillInDTypeGivenDTypeAndDataType(knowledge, op.dtype(), dataType.getValue());
```